app/scripts/lib/validate.js

```diff
@@ -13,6 +13,38 @@ define(function (require, exports, module) {
   // taken from the fxa-auth-server
   var HEX_STRING = /^(?:[a-fA-F0-9]{2})+$/;
 
+  // URL RegEx taken from http://blog.mattheworiordan.com/post/13174566389/url-regular-expression-for-links-with-or-without
+  // jscs:disable maximumLineLength
+  var urlRegEx = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)/;
```

Comment:
Thanks @vladikoff - @rfk and I had a chat today about tightening up the requirements for a `redirect_uri` until specifically requested. A reasonable base level restriction is to require either `https://` and `urn:` unless running locally. We can loosen the restriction if required. This would require updates to the oauth server and possibly oauth console.

@vbudhram - keep what you have now, I opened https://github.com/mozilla/fxa-oauth-server/issues/388 and we can update the content server when that issue is handled.